src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -453,7 +453,8 @@ internal static bool IsEdgeSupportedType(Type type)
             {
                 // since HTTP headers are case insensitive, we lower-case the keys
                 // as does Node.js request object
-                headers.Add(header.Key.ToLowerInvariant(), header.Value.First());
+                string values = string.Join(" ", header.Value);
```

Comment:
request.Headers.ToString() might give you what you're looking for (may need to do some parsing)